From 2aca9b24ae5fbddb650ea771fb5b8c2d856f9fcd Mon Sep 17 00:00:00 2001
From: Laszlo Ersek <lersek@redhat.com>
Date: Tue, 17 Jun 2014 13:11:41 +0200
Subject: [PATCH 01/44] dump: Drop pointless error_is_set(), DumpState member errp

RH-Author: Laszlo Ersek <lersek@redhat.com>
Message-id: <1403010708-7504-2-git-send-email-lersek@redhat.com>
Patchwork-id: 59250
O-Subject: [RHEL-6.6 qemu-kvm PATCH 1/8] dump: Drop pointless error_is_set(), DumpState member errp
Bugzilla: 1102659
RH-Acked-by: Luiz Capitulino <lcapitulino@redhat.com>
RH-Acked-by: Dr. David Alan Gilbert (git) <dgilbert@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

In qmp_dump_guest_memory(), the error must be clear on entry, and we
always bail out after setting it, directly or via dump_init().
Therefore, both error_is_set() are always false.  Drop them.

DumpState member errp is now write-only.  Drop it, too.

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Reviewed-by: Eric Blake <eblake@redhat.com>
Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
(cherry picked from commit 66ef8bd9c16b547c985cbe7468dcf60280c993eb)
Signed-off-by: Laszlo Ersek <lersek@redhat.com>
---
 dump.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 dump.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/dump.c b/dump.c
index ac991e3..bb320c1 100644
--- a/dump.c
+++ b/dump.c
@@ -91,7 +91,6 @@ typedef struct DumpState {
     bool has_filter;
     int64_t begin;
     int64_t length;
-    Error **errp;
 
     uint8_t *note_buf;          /* buffer for notes */
     size_t note_buf_offset;     /* the writing place in note_buf */
@@ -1569,7 +1568,6 @@ static int dump_init(DumpState *s, int fd, bool has_format,
         nr_cpus++;
     }
 
-    s->errp = errp;
     s->fd = fd;
     s->has_filter = has_filter;
     s->begin = begin;
@@ -1776,11 +1774,11 @@ void qmp_dump_guest_memory(bool paging, const char *file, bool has_begin,
     }
 
     if (has_format && format != DUMP_GUEST_MEMORY_FORMAT_ELF) {
-        if (create_kdump_vmcore(s) < 0 && !error_is_set(s->errp)) {
+        if (create_kdump_vmcore(s) < 0) {
             error_set(errp, QERR_IO_ERROR);
         }
     } else {
-        if (create_vmcore(s) < 0 && !error_is_set(s->errp)) {
+        if (create_vmcore(s) < 0) {
             error_set(errp, QERR_IO_ERROR);
         }
     }
-- 
1.7.1

